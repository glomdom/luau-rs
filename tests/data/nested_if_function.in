fn main(a: i32, b: i32) -> bool {
    if a == 1 {
        if b == 2 {
            true
        } else if b == 3 {
            true
        } else {
            false
        }
    } else if a == 2 {
        true
    } else if b == 3 {
        if a == 3 {
            true
        } else {
            false
        }
    } else {
        false
    }
}