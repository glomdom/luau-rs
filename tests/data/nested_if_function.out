Block {
    statements: [
        Function {
            name: "main",
            params: [
                LuauParam {
                    name: "a",
                    typ: "number",
                    is_ref: false,
                },
            ],
            ret_type: Some(
                LuauType {
                    type_name: "boolean",
                },
            ),
            body: Block {
                statements: [
                    If {
                        condition: BinaryOp {
                            op: "==",
                            left: Value(
                                "a",
                            ),
                            right: Value(
                                "1",
                            ),
                        },
                        then_branch: Block {
                            statements: [
                                Return {
                                    value: Some(
                                        Value(
                                            "true",
                                        ),
                                    ),
                                },
                            ],
                        },
                        else_branch: Some(
                            If {
                                condition: BinaryOp {
                                    op: "==",
                                    left: Value(
                                        "a",
                                    ),
                                    right: Value(
                                        "2",
                                    ),
                                },
                                then_branch: Block {
                                    statements: [
                                        Return {
                                            value: Some(
                                                Value(
                                                    "false",
                                                ),
                                            ),
                                        },
                                    ],
                                },
                                else_branch: Some(
                                    Block {
                                        statements: [
                                            Return {
                                                value: Some(
                                                    Value(
                                                        "true",
                                                    ),
                                                ),
                                            },
                                        ],
                                    },
                                ),
                            },
                        ),
                    },
                ],
            },
        },
    ],
}